state-sync/state-sync-v2/data-streaming-service/src/data_stream.rs

```diff
@@ -79,7 +79,7 @@ pub struct DataStream<T> {
     notifications_to_responses: BTreeMap<NotificationId, ResponseContext>,
 
     // The channel on which to send data notifications when they are ready.
-    notification_sender: channel::aptos_channel::Sender<(), DataNotification>,
+    notification_sender: mpsc::Sender<DataNotification>,
```

Comment:
so the main change is channel::aptos_channel to mpsc::Sender, everything else is just async noise.

Can you explain the meaning of the difference to a noob here? :)

Reply:
@igor-aptos, yeah. `channel::aptos_channel` doesn't block when the channel gets full. New messages are just dropped, but mpsc::Sender actually blocks the sender 😄

In this context, it relates to the execute-commit pipeline. We have a driver sending notifications to the executor, and the executor sending notifications to committer. Notification sending happens via these channels. Channels have bounds (i.e., sizes of `X`). When the channel is full, we were dropping messages and throwing errors, which was causing the driver to wait for the pipeline to drain, before restarting the sync progress. Instead, we're gonna block until the channel is no longer full. It won't make things faster if the bottleneck is the execute-commit pipeline, but it'll hopefully help smooth out performance. Does that make sense? 🤔